### PR TITLE
Rewrite to behave as extension of current classes

### PR DESCRIPTION
**Pull request by migl**

### CODE REVIEW

File: lib/gestalt/store.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    # Stores a value in the passed key
+    #
+    # @param key the target key
+    # @param value the value to store
+    def []=(key, value)
+      @configuration[key] = value
+    end
+
+    def breadcrumbs
```
Comment (cdotta) [inline lib/gestalt/store.rb:42]: doc this one

File: lib/gestalt.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      end
+    end
+
+    yield if block_given?
+  end
+
+  private
+
+  def self._gestalt_init(base)
+    # Initialize with default values
```
Comment (cdotta) [inline lib/gestalt.rb]: I think is better to add method documentation than inline documentation

File: lib/gestalt.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    # Make sure to change these AFTER the module has been included/extended
+    config = Store.new({}, 'gestalt')
+    config.config_path = CONFIG_PATH
+    config.ignore_unsupported_extensions = IGNORE_UNSUPPORTED_EXTENSIONS
+
+    # "config" is only exposed through this closure
+    base.define_singleton_method(:gestalt) do |&block|
+      if block
```
Comment (cdotta) [inline lib/gestalt.rb]: block ? block.call(config) : config

File: lib/gestalt.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      end
+    end
+  end
+
+  def _gestalt
+    self.respond_to?(:gestalt) ? self.gestalt : self.class.gestalt
+  end
+
+  def _gestalt_parse_file(file)
+    extension = File.extname(file)
```
Comment (cdotta) [inline lib/gestalt.rb:95]: maybe is a good idea to abstract the file concept in a class, and then just use load by duck typing

Comment (migl) [inline lib/gestalt.rb:95]: Future version, maybe?

**Comment by cdotta** [comment]
Good for me!

**Pull request merged**